### PR TITLE
Add Itoa/Atoi overloads, add string-related methods to Std contract

### LINKED ISSUE #1928: Add Itoa/Atoi overloads and memory/string stdlib methods
See neo-project/neo#2416.

### PR DESCRIPTION
**Pull request by fyrchik**
Close #1928 .

### CODE REVIEW

**Review by AnnaShaleva — APPROVED** [review]

File: pkg/interop/native/std/std.go [hunk 1]
```diff
@@ -146,3 +146,18 @@ func MemorySearchFull(mem, pattern []byte, start int, backward bool) int {
...
+// It uses `stringSplit` method of StdLib native contract.
+func StringSplit(s, sep string) []string {
+	return contract.Call(interop.Hash160(Hash), "stringSplit", contract.NoneFlag,
+		s, sep).([]string)
+}
+
+// StringSplitNonEmpty splits s by ocurrences of sep.
+// If removeEmpty is true, remove empty items from the resulting slice.
+// It uses `stringSplit` method of StdLib native contract.
+func StringSplitNonEmpty(s, sep string, removeEmpty bool) []string {
```
Comment (roman-khimov) [inline pkg/interop/native/std/std.go:160]: Maybe we can drop `removeEmpty` parameter here? `StringSplit` will do what it does and `StringSplitNonEmpty` will just call add `true` to parameter list?

File: pkg/interop/native/std/std.go [hunk 2]
```diff
@@ -121,3 +121,28 @@ func MemoryCompare(s1, s2 []byte) int {
...
+func MemorySearchIndex(mem, pattern []byte, start int) int {
+	return contract.Call(interop.Hash160(Hash), "memorySearch", contract.NoneFlag,
+		mem, pattern, start).(int)
+}
+
+// MemorySearchFull returns index of the first occurence of val in mem starting from start.
+// If backward is true, search backwards, start index is not included.
+// If not found, -1 is returned.
+// It uses `memorySearch` method of StdLib native contract.
+func MemorySearchFull(mem, pattern []byte, start int, backward bool) int {
```
Comment (roman-khimov) [inline pkg/interop/native/std/std.go:145]: Same thing here, we can do `MemorySearchBackward` and pass `true`, other variants are implicitly `false` anyway.

File: pkg/interop/native/std/std.go (hunk 1 above)
Comment (fyrchik) [reply pkg/interop/native/std/std.go:160]: I agree that the interface will be cleaner. Here I have followed the policy "of 1-1 wrappers".

File: pkg/interop/native/std/std.go [hunk 3]
```diff
@@ -121,3 +121,28 @@ func MemoryCompare(s1, s2 []byte) int {
...
+// It uses `memorySearch` method of StdLib native contract.
+func MemorySearch(mem, pattern []byte) int {
+	return contract.Call(interop.Hash160(Hash), "memorySearch", contract.NoneFlag,
+		mem, pattern).(int)
+}
+
+// MemorySearchIndex returns index of the first occurence of val in mem starting from start.
+// If not found, -1 is returned.
+// It uses `memorySearch` method of StdLib native contract.
+func MemorySearchIndex(mem, pattern []byte, start int) int {
```
[the remark this answers is not in the document]
Comment (fyrchik) [reply pkg/interop/native/std/std.go:136]: `Index`/`LastIndex` is closer to std-lib `bytes` and `strings` packages.

[the remark this answers is not in the document]
Comment (fyrchik) [reply pkg/interop/native/std/std.go:136]: I also think `BytesIndex` is more readable.

File: pkg/interop/native/std/std.go [hunk 4]
```diff
@@ -93,9 +93,71 @@ func Itoa(num int, base int) string {
...
+// MemorySearchIndex returns index of the first occurrence of val in mem starting from start.
+// If not found, -1 is returned.
+// It uses `memorySearch` method of StdLib native contract.
+func MemorySearchIndex(mem, pattern []byte, start int) int {
+	return contract.Call(interop.Hash160(Hash), "memorySearch", contract.NoneFlag,
+		mem, pattern, start).(int)
+}
+
+// MemorySearchLastIndex returns index of the last occurrence of val in mem ending before start.
+// If backward is true, search backwards, start index is not included.
```
Comment (roman-khimov) [inline pkg/interop/native/std/std.go:142]: What `backward`?

File: pkg/interop/native/std/std.go [hunk 5]
```diff
@@ -93,9 +93,71 @@ func Itoa(num int, base int) string {
...
+
+// StringSplit splits s by occurrences of sep.
+// It uses `stringSplit` method of StdLib native contract.
+func StringSplit(s, sep string) []string {
+	return contract.Call(interop.Hash160(Hash), "stringSplit", contract.NoneFlag,
+		s, sep).([]string)
+}
+
+// StringSplitNonEmpty splits s by occurrences of sep.
+// If removeEmpty is true, remove empty items from the resulting slice.
```
Comment (roman-khimov) [inline pkg/interop/native/std/std.go:158]: What `removeEmpty`?

**Review by roman-khimov — APPROVED** [review]

**Pull request merged**